New Indonesian Translation: React Developer Tools (#334)
reactjs/id.react.dev: src/content/learn/react-developer-tools.md

---

> title: Alat pengembang React

Reviewer: ```suggestion
title: React Developer Tools
```

Best not to translate this since it references the name of the React Developer Tools extension.

---

<Intro>

> Gunakan Alat Pengembang React untuk memeriksa [komponen](/learn/your-first-component), menyunting [*props*](/learn/passing-props-to-a-component) dan [*state*](/learn/state-a-components-memory), dan mengidentifikasi masalah kinerja.

Reviewer: ```suggestion
Gunakan React Developer Tools untuk memeriksa [komponen](/learn/your-first-component), menyunting [*props*](/learn/passing-props-to-a-component) dan [*state*](/learn/state-a-components-memory), dan mengidentifikasi masalah kinerja.
```

Idem.

</Intro>

<YouWillLearn>

> * Cara memasang Alat Pengembang React

Reviewer: ```suggestion
* Cara menginstal React Developer Tools
```

install -> instal

Please check the [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium).

</YouWillLearn>

> ## Ekstensi Browser {/*browser-extension*/}

Reviewer: ```suggestion
## Ekstensi Peramban {/*browser-extension*/}
```

browser -> peramban

> Cara termudah untuk melakukan debugging website yang dibangun dengan React adalah dengan memasang ekstensi browser Alat Pengembang React. Ekstensi ini tersedia untuk beberapa browser populer:

Reviewer: ```suggestion
Cara termudah untuk melakukan debugging website yang dibangun dengan React adalah dengan memasang ekstensi peramban React Developer Tools. Ekstensi ini tersedia untuk beberapa browser populer:
```

Idem.

> * [Pasang untuk **Chrome**](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en)
> * [Pasang untuk **Firefox**](https://addons.mozilla.org/en-US/firefox/addon/react-devtools/)
> * [Pasang untuk **Edge**](https://microsoftedge.microsoft.com/addons/detail/react-developer-tools/gpphkfbcpidddadnkolkpfckpihlkkil)

Reviewer: ```suggestion
* [Instal untuk **Chrome**](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en)
* [Instal untuk **Firefox**](https://addons.mozilla.org/en-US/firefox/addon/react-devtools/)
* [Instal untuk **Edge**](https://microsoftedge.microsoft.com/addons/detail/react-developer-tools/gpphkfbcpidddadnkolkpfckpihlkkil)
```

Idem

Sekarang, jika Anda mengunjungi sebuah website **yang dibangun dengan React,** Anda akan melihat panel _Components_ dan _Profiler_.

![React Developer Tools extension](/images/docs/react-devtools-extension.png)

> ### Safari and browser lainnya {/*safari-and-other-browsers*/}
> Untuk browser lainnya (misalnya, Safari), pasang paket npm [`react-devtools`](https://www.npmjs.com/package/react-devtools):

Reviewer: ```suggestion
### Safari and peramban lainnya {/*safari-and-other-browsers*/}
Untuk peramban lainnya (misalnya, Safari), pasang package npm [`react-devtools`](https://www.npmjs.com/package/react-devtools):
```

Reviewer 2: `pasang package` or `instal paket`, according to the [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#i)?

Author: instal _package_ according to Glossary

Reviewer 2: `instal package` it is then

Author: I already push new commit that match with Glossarium. Take a look [here](https://github.com/reactjs/id.react.dev/pull/334/commits/b75c6f8898717cd2dfeae7a0eba510fa8e91f343)

<Intro>

Gunakan React Developer Tools untuk memeriksa [komponen](/learn/your-first-component), menyunting [*props*](/learn/passing-props-to-a-component) dan [*state*](/learn/state-a-components-memory), dan mengidentifikasi masalah kinerja.

</Intro>

<YouWillLearn>

> * Cara menginstal React Developer Tools

Reviewer 3: Cara memasang

Reviewer 2: According to the [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#i), it's fine to use `menginstal`

Reviewer: Yep, let's keep it `instal`

Reviewer 3: @resir014 @mhaidarhanif Nice, need to check this one also, belum menggunakan `instal` https://github.com/reactjs/id.react.dev/pull/333/files

</YouWillLearn>

## Ekstensi Peramban {/*browser-extension*/}

Cara termudah untuk melakukan debugging website yang dibangun dengan React adalah dengan memasang ekstensi peramban React Developer Tools. Ekstensi ini tersedia untuk beberapa peramban populer:

> * [Instal untuk **Chrome**](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en)

Reviewer 2: According to the [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#i), it's fine to use `Instal`

> * [Instal untuk **Chrome**](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en)

Reviewer: Yep, let's keep it `instal`

> * [Instal untuk **Chrome**](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en)

Reviewer 3: @resir014 @mhaidarhanif Nice, need to check this one also, belum menggunakan `instal` https://github.com/reactjs/id.react.dev/pull/333/files

* [Instal untuk **Firefox**](https://addons.mozilla.org/en-US/firefox/addon/react-devtools/)
* [Instal untuk **Edge**](https://microsoftedge.microsoft.com/addons/detail/react-developer-tools/gpphkfbcpidddadnkolkpfckpihlkkil)

Sekarang, jika Anda mengunjungi sebuah website **yang dibangun dengan React,** Anda akan melihat panel _Components_ dan _Profiler_.

![React Developer Tools extension](/images/docs/react-devtools-extension.png)

### Safari and Peramban lainnya {/*safari-and-other-browsers*/}

> Untuk peramban lainnya (misalnya, Safari), instal paket npm [`react-devtools`](https://www.npmjs.com/package/react-devtools):

Reviewer 2: According to the [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#i), it's fine to use `instal paket`

> Untuk peramban lainnya (misalnya, Safari), instal paket npm [`react-devtools`](https://www.npmjs.com/package/react-devtools):

Reviewer 3: @resir014 @mhaidarhanif Nice, need to check this one also, belum menggunakan `instal` https://github.com/reactjs/id.react.dev/pull/333/files

npm install -g react-devtools
```

Selanjutnya buka alat pengembang dari terminal:

```bash
react-devtools
```

Lalu sambungkan website Anda dengan menambahkan tag `<script>` berikut ke awal `<head>` website Anda:

```html {3}
<html>
  <head>
    <script src="http://localhost:8097"></script>
```

Reload website Anda sekarang untuk melihatnya di alat pengembang.

![React Developer Tools standalone](/images/docs/react-devtools-standalone.png)

## Mobile (React Native) {/*mobile-react-native*/}

> Alat Pengembang React dapat digunakan untuk memeriksa aplikasi yang dibangun dengan [React Native](https://reactnative.dev/) juga.

Reviewer: ```suggestion
React Developer Tools dapat digunakan untuk memeriksa aplikasi yang dibangun dengan [React Native](https://reactnative.dev/) juga.
```

> Cara termudah untuk menggunakan Alat Pengembang React adalah dengan memasangnya secara global:

Reviewer: ```suggestion
Cara termudah untuk menggunakan React Developer Tools adalah dengan memasangnya secara global:
```

npm install -g react-devtools
```

Selanjutnya buka alat pengembang dari terminal:
```bash
react-devtools
```

> Alat Pengembang React akan terhubung ke aplikasi React Native lokal yang sedang berjalan.

Reviewer: ```suggestion
React Developer Tools akan terhubung ke aplikasi React Native lokal yang sedang berjalan.
```
